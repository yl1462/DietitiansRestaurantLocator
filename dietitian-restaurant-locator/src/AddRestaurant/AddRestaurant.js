import React, { Component } from 'react';
import config from '../config';

class AddRestaurant extends Component {

  state = {
    the_restaurant: '',
    type: ''
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  // posting new restaurant to the page and database
  handleSubmit = (e) => {
    e.preventDefault()
    fetch(`${config.API_ENDPOINT}/api/restaurant`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        the_restaurant: this.state.the_restaurant,
        type: this.state.type,
      })
    })
      .then(res => {
        if (!res.ok) {
          return res.json().then(err => {
            console.log(`Error Message: ${err}`)
            throw err
          })
        }
        return res.json()
      })

      // add restaurant, then go back to homepage
      .then(restaurant => {
        this.props.addRestaurant(restaurant)
        this.props.history.push('/home')
      })
      .catch(err => {
        this.setState({ err })
      })
  }

  render() {
    let { the_restaurant, type } = this.props

    return (
      <>
        <div className="AddRestaurant">
          <form onSubmit={this.handleSubmit}>

            {/* must give a title to the new restaurant */}
            <input
              type='text'
              placeholder='Name of the Restaurant'
              value={the_restaurant}
              onChange={this.handleChange}
              name='the_restaurant'
              className="Placeholder"
              required
              pattern="[A-Za-z0-9\~\!\@\#\$\%\^\*\(\)\_]{3}"
              the_restaurant="at least 3 characters required"
            />
            <br />

            {/* optional description */}
            <select
              required
              onChange={this.handleChange}
              name='type'
              className="Placeholder"
              type='selected'
              value={type}
            >
              <option value=''>---Type of Diet---</option>
              <option value='Keto'>Keto</option>
              <option value='Mediterranean'>Mediterranean</option>
              <option value='Plant-based'>Plant-based</option>
            </select>
            <br />

            {/* submit the new restaurant, then go back to homepage */}
            <button type='submit' className='AddRestaurant-button'>Submit</button>
          </form>
        </div>
      </>
    );
  }
}

export default AddRestaurant;