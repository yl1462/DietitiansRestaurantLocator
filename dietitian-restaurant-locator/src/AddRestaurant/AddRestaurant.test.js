import React from 'react';
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddRestaurant from './AddRestaurant'

describe('renders without crashing', () => {
  //test if page loads
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <AddRestaurant />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe(`AddRestaurant component`, () => {
  const props = {
    the_restaurant: 'McDonald',
    type: 'keto'
  }

  //test if testing new restaurant could be added
  it('renders the new restaurant given props', () => {
    const wrapper = shallow(<AddRestaurant {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

