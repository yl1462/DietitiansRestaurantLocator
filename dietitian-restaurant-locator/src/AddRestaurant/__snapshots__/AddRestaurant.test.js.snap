// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRestaurant component renders the new restaurant given props 1`] = `
<Fragment>
  <div
    className="AddRestaurant"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="Placeholder"
        name="the_restaurant"
        onChange={[Function]}
        pattern="[A-Za-z0-9\\\\~\\\\!\\\\@\\\\#\\\\$\\\\%\\\\^\\\\*\\\\(\\\\)\\\\_]{3}"
        placeholder="Name of the Restaurant"
        required={true}
        the_restaurant="at least 3 characters required"
        type="text"
        value="McDonald"
      />
      <br />
      <select
        className="Placeholder"
        name="type"
        onChange={[Function]}
        required={true}
        type="selected"
        value="keto"
      >
        <option
          value=""
        >
          ---Type of Diet---
        </option>
        <option
          value="Keto"
        >
          Keto
        </option>
        <option
          value="Mediterranean"
        >
          Mediterranean
        </option>
        <option
          value="Plant-based"
        >
          Plant-based
        </option>
      </select>
      <br />
      <button
        className="AddRestaurant-button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</Fragment>
`;
